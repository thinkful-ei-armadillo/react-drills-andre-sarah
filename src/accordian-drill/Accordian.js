import React, { Component } from 'react';

class Accordian extends Component{
    state ={
        paragraph: <p /> ,
        selected: -1
    }
    
    handleButtonClicked = (index) => {
        console.log(`button clicked`)
        this.setState({selected: index})
        console.log(index);
        console.log({selected: index})
        
    }

    render() {
        return (
            <div><ul>
        {this.props.sections.map((item, index) => <li key={index}><button type="button" onClick={() => this.handleButtonClicked(index)}>{item.title} </button> {this.state.selected === index ? this.props.sections[index].content : ''} </li>)}
        
            </ul>
            </div>
        )
    }
}

export default Accordian;