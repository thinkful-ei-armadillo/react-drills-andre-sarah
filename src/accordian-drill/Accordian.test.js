import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Accordian from './Accordian';
import { sections } from '../store';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

describe('Testing accordian', () => {
  //snapshot test
  it('component renders an empy li when the sections prop is not supplied', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Accordian sections={sections} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('component renders no sections as active by default', () => {
    const tree = renderer
      .create(<Accordian sections={sections} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('component opens a clicked section', () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    wrapper.find('button').at(1).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('component only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordian sections={sections} />);
    wrapper.find('button').at(1).simulate('click');
    wrapper.find('button').at(2).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});