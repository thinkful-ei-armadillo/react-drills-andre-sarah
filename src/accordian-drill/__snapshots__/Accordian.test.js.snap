// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing accordian component only opens the last section when multiple sections have been clicked 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
         
      </button>
       
       
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
         
      </button>
       
       
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
         
      </button>
       
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
       
    </li>
  </ul>
</div>
`;

exports[`Testing accordian component opens a clicked section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
         
      </button>
       
       
    </li>
    <li
      key="1"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
         
      </button>
       
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
       
    </li>
    <li
      key="2"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
         
      </button>
       
       
    </li>
  </ul>
</div>
`;

exports[`Testing accordian component renders no sections as active by default 1`] = `
<div>
  <ul>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        Section 1
         
      </button>
       
      
       
    </li>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        Section 2
         
      </button>
       
      
       
    </li>
    <li>
      <button
        onClick={[Function]}
        type="button"
      >
        Section 3
         
      </button>
       
      
       
    </li>
  </ul>
</div>
`;
